File: frontend/src/api/incidents.ts
import { api } from "./client";

export interface IncidentTimelineEntry {
  timestamp: string;
  message: string;
  author: string;
}

export interface Incident {
  id: string;
  title: string;
  service: string;
  severity: "SEV1" | "SEV2" | "SEV3";
  status: string;
  opened_at: string;
  closed_at?: string;
  timeline: IncidentTimelineEntry[];
  runbook_url?: string;
  summary?: string;
}

export interface IncidentResponse {
  incident: Incident;
  recommendations: string[];
}

export interface IncidentCreate {
  title: string;
  service: string;
  severity: "SEV1" | "SEV2" | "SEV3";
  description: string;
}

export const fetchIncidents = async (): Promise<Incident[]> => {
  const { data } = await api.get<Incident[]>("/incidents/");
  return data;
};

export const fetchIncident = async (id: string): Promise<IncidentResponse> => {
  const { data } = await api.get<IncidentResponse>(`/incidents/${id}`);
  return data;
};

export const createIncident = async (
  payload: IncidentCreate
): Promise<IncidentResponse> => {
  const { data } = await api.post<IncidentResponse>("/incidents/", payload);
  return data;
};

File: backend/app/schemas/incidents.py
from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime
from enum import Enum
from typing import List, Optional


class Severity(str, Enum):
    sev1 = "SEV1"
    sev2 = "SEV2"
    sev3 = "SEV3"


@dataclass
class IncidentTimelineEntry:
    timestamp: datetime
    message: str
    author: str


@dataclass
class Incident:
    id: str
    title: str
    service: str
    severity: Severity
    status: str
    opened_at: datetime
    closed_at: Optional[datetime] = None
    timeline: List[IncidentTimelineEntry] = field(default_factory=list)
    runbook_url: Optional[str] = None
    summary: Optional[str] = None


@dataclass
class IncidentCreate:
    title: str
    service: str
    severity: Severity
    description: str


@dataclass
class IncidentResponse:
    incident: Incident
    recommendations: List[str]
